src/analysis/vector/geometry_checker/qgsgeometryselfintersectioncheck.cpp

```diff
@@ -114,7 +114,8 @@ void QgsGeometrySelfIntersectionCheck::fixError( QgsGeometryCheckError *error, i
   QgsPoint p2 = geom->vertexAt( QgsVertexId( vidx.part, vidx.ring, ( inter.segment1 + 1 ) % nVerts ) );
   QgsPoint q2 = geom->vertexAt( QgsVertexId( vidx.part, vidx.ring, ( inter.segment2 + 1 ) % nVerts ) );
   QgsPoint s;
-  if ( !QgsGeometryUtils::segmentIntersection( p1, p2, q1, q2, s, mContext->tolerance ) )
+  bool intersection;
```

Comment:
Initialise also. It makes me nervous seeing uninitialised variables like this, even if they are unused. Plus it makes code analysers freak out.